### PR TITLE
Before deleting data source modal window will show apps where this data source is used

### PR DESCRIPTION
**Pull request by MaksymShokin**
## Issue
Fliplet/fliplet-studio#3774

## Description
Replaced default confirm with Fliplet.confirm. Added names of apps where the data source is used.

## Screenshots/screencasts
deleting demo

## Backward compatibility
This change is fully backward compatible.

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -536,9 +536,27 @@ $('#app')
...
-    var confirmAlert = confirm('Are you sure you want to delete this data source? All entries will be deleted.');
+    var message = 'Are you sure you want to delete this data source? All entries will be deleted.';
+    var currentDS = _.find(dataSources, function(ds) {
+      return ds.id === currentDataSourceId;
+    });
+
+    if (currentDS.apps.length) {
+      var appPrefix = currentDS.apps.length > 1 ? 'apps: ' : 'app: ';
+      var appUsedIn = currentDS.apps.map(function(elem) {
+        return ' ' + elem.name;
```
Comment (squallstar) [inline js/interface.js]: just return `elem.name` and then join the results with `.join(', ')`

File: js/interface.js [hunk 2]
```diff
@@ -536,9 +536,27 @@ $('#app')
...
+      });
+
+      message = 'Are you sure you want to delete this data source? The data source is used in the next ' + appPrefix + appUsedIn + '. All entries will be deleted.';
```
Comment (squallstar) [inline js/interface.js]: the `Are you sure you want to delete this data source` is repeated twice, please refactor to only append the relevant part.

Comment (squallstar) [inline js/interface.js]: "The data source is currently in use by the following apps: A, B, C."

Comment (MaksymShokin) [reply js/interface.js]: Fixed

Comment (MaksymShokin) [reply js/interface.js]: Fixed

File: js/interface.js (hunk 1 above)
Comment (MaksymShokin) [reply js/interface.js]: Fixed

**Pull request merged**